tapir/shifts/views/views.py

```diff
@@ -353,3 +357,31 @@ def get(self, request, *args, **kwargs):
         call_command("run_freeze_checks")
         messages.info(request, _("Frozen statuses updated."))
         return super().get(request, args, kwargs)
+
+
+# class WatchShiftView(LoginRequiredMixin, RedirectView):
+#     def post(self, request, *args, **kwargs):
+#         shift = get_object_or_404(Shift, pk=kwargs["shift_id"])
+#         ShiftWatch.objects.get_or_create(user=request.user, shift=shift)
+#         return redirect("shifts:shift_detail", pk=kwargs["shift_id"])
+#
+
+
+class UnwatchShiftView(LoginRequiredMixin, RedirectView):
+    def post(self, request, *args, **kwargs):
+        shift = get_object_or_404(Shift, id=kwargs["shift"])
+        ShiftWatch.objects.filter(user=request.user, shift=shift).delete()
+        return redirect("shifts:shift_detail", pk=kwargs["shift"])
+
+
+class WatchShiftView(LoginRequiredMixin, TapirFormMixin, CreateView):
```

Comment:
The page title and the card title are missing, see an example in `tapir.coop.views.draftuser.DraftUserCreateView.get_context_data`